src/recorder/jamcontroller.cpp

```diff
@@ -77,19 +78,29 @@ void CJamController::SetRecordingDir ( QString newRecordingDir,
                                        int     iServerFrameSizeSamples,
                                        bool    bDisableRecording )
 {
-    if ( bRecorderInitialised && pthJamRecorder != nullptr )
+    if ( bRecorderInitialised )
     {
-        // We have a thread and we want to start a new one.
-        // We only want one running.
-        // This could take time, unfortunately.
-        // Hopefully changing recording directory will NOT happen during a long jam...
-        emit EndRecorderThread();
-        pthJamRecorder->wait();
-        pthJamRecorder = nullptr;
+        if ( pthJamRecorder != nullptr )
+        {
+            // We have a thread and we want to start a new one.
+            // We only want one running.
+            // This could take time, unfortunately.
+            // Hopefully changing recording directory will NOT happen during a long jam...
+            emit EndRecorderThread();
+            pthJamRecorder->wait();
+            delete pthJamRecorder;
+            pthJamRecorder = nullptr;
+        }
     }
 
     if ( !newRecordingDir.isEmpty() )
     {
+        if ( pJamRecorder != nullptr )
+        {
+            // We have a recorder running already. Terminate it.
```

Comment:
Just so I don't get so nervous reading this, can it be a clearer comment, please:
`// We have a reference to a CJamRecorder instance that should now have finished.  Clean up the instance.`
(`pJamRecorder->moveToThread ( pthJamRecorder );` puts `pJamRecorder` into the thread and the `pthJamRecorder ->wait()` above - if we had one - waited for it to finish.)